tests/acceptance/async_test_base.py

```diff
@@ -33,11 +31,29 @@ class AsyncTestCase(unittest.TestCase):
 
     def setUp(self):
         self.logger = logging.getLogger(self.__class__.__name__)
-        self.parameters = pika.URLParameters(
-            'amqp://guest:guest@localhost:5672/%2F')
+        self.parameters = pika.ConnectionParameters(
+            host='localhost',
+            port=5672)
+        if self.should_test_tls():
+            self.logger.info('testing using TLS/SSL connection to port 5671')
+            self.parameters.port = 5671
+            self.parameters.ssl = True
+            self.parameters.ssl_options = dict(
+                ssl_version=ssl.PROTOCOL_TLSv1,
+                ca_certs="testdata/certs/ca_certificate.pem",
+                keyfile="testdata/certs/client_key.pem",
+                certfile="testdata/certs/client_certificate.pem",
+                cert_reqs=ssl.CERT_REQUIRED)
```

Comment:
Please make `should_test_tls` and `ssl_options` available to other tests. Here is a proposal:

1. Use an environment variable (e.g., `PIKA_TEST_SSL_OPTIONS_FILE`) to store the path of the JSON file containing the settings for assigning to `self.parameters.ssl_options`. This way, developers can set this up on their test environments as needed.
2. Expose a function via `test_utils.py` with the functionality of should_test_tls.
3. Expose a function via `test_utils.py` for loading the `ssl_options` from the file referenced by the corresponding (e.g., `PIKA_TEST_SSL_OPTIONS_FILE`) environment variable.
4. Use both functions in this test instead of hard-coding them.

Reply:
> Please make should_test_tls and ssl_options available to other tests

There aren't other tests that need them at this point, right? I don't really want to expand this configuration beyond what is needed by our continuous integration environments in this PR. I will, however, add a section in the README giving instructions on how to use the included certs if someone wants to run TLS tests themselves. Since we include a `rabbitmq.conf.in` file it is pretty easy to get going.

Reply:
@lukebakken, I was hoping to extend SSL testing to BlockingConnection and the tests for the new SSL/transport refactoring work that I am doing. In addition, my transport tests will need the location of the server key file. That said, we can wait to add this functionality in the next "sprint".

Reply:
OK let's do it as part of that work.